src/clj/api/server.clj

```diff
@@ -25,6 +25,50 @@
 
 (declare server)
 
+;; NOTE: this is not ideal as every copy of the ECS service will maintain it's own copy
+;; It would be best to persist it to the RDB
+;; however as we are currently running only one task per service it will suffice
+(def jailed-ips (atom {}))
+
+(defn update-ip [ip timestamp]
+  (swap! jailed-ips update-in [ip :timestamps] conj timestamp))
+
+(defn is-jailed? [ip now]
+  (let [info       (@jailed-ips ip)
+        timestamps (:timestamps info)]
+    (and info
+         ;; 5 times
+         (> (count timestamps) 4)
+         ;; within 10 minutes
+         (< (- now (last timestamps)) (* 10 60 1000))
+         (not (:jail-lift-time info))
+         ;; jailed for 20 minutes
+         (swap! jailed-ips update-in [ip :jail-lift-time] (fn [_] (+ now (* 20 60 1000)))))))
+
+(defn is-temporarily-jailed? [ip now]
```

Comment:
This one the same, I would choose something more like `check-and-lift` with a less smart condition like :

```clojure
(when (and info
           jail-lift-time
           (> now jail-lift-time))
  (swap! jailed-ips update-in [ip] dissoc :jail-lift-time))
```

Reply:
I admit I whipped it late in the evening and I actually don't like it that much today :wink: 
It conflates pure and side-effects is hard to read & understand - I think it could be implemented as a state machine.

I think what I'll do is we keep this handy if we need to deploy a solution ASAP, but I'll create another branch on top of it and )time permitting) I will make something better in a few days. WYT?

Reply:
I think it is a good idea yeah, if the attack stopped then why hurry

Reply:
resolving as outdated